### PR TITLE
3l-offZ event categories division update

### PR DESCRIPTION
**Pull request by ywan2**
This PR is crucial for the update of analysis including the division of 3l event categories.

The changes are made in `event_selection.py` to add an extra masking for low mass region of 3l-offZ channels (to use in the analysis processor in topeft).

### CODE REVIEW

**Review by bryates — COMMENTED** [review]
Looks good to me. I'm happy merging this into this repo, but @kmohrman does this line up with your refactoring plans? It's _sort_ of analysis specific, but at the same time, it's general enough that anyone can use it.

File: topcoffea/modules/event_selection.py [hunk 1]
```diff
@@ -69,3 +69,19 @@ def get_Z_peak_mask(lep_collection,pt_window,flavor="os",zmass=91.2):
         raise Exception(f"Error: flavor requirement \"{flavor}\" is unknown.")
     sfosz_mask = ak.flatten(ak.any((zpeak_mask & sf_mask),axis=1,keepdims=True)) # Use flatten here because it is too nested (i.e. it looks like this [[T],[F],[T],...], and want this [T,F,T,...]))
     return sfosz_mask
+
+# Returns a mask for all events with any os lepton pair within low region of off-Z
+# Mask will be True if any combination of 2 leptons from within the given collection satisfies the requirement
+def get_off_Z_mask_low(lep_collection,pt_window,flavor="os"):
```
Comment (bryates) [inline topcoffea/modules/event_selection.py:75]: If we want to generalize this a bit, we could remove `_low` from the name, and change `pt_window` to a string that takes `"low"` or `"high"`

File: topcoffea/modules/event_selection.py [hunk 2]
```diff
@@ -69,3 +69,19 @@ def get_Z_peak_mask(lep_collection,pt_window,flavor="os",zmass=91.2):
...
+    ll_pairs = ak.combinations(lep_collection, 2, fields=["l0","l1"])
+    zpeak_mask = ((91.2-(ll_pairs.l0+ll_pairs.l1).mass)>pt_window)
```
Comment (bryates) [inline topcoffea/modules/event_selection.py:77]: _if_ we generalize this, then the `pt_window` comparison would change.

**Comment by kmohrman** [comment]
I agree that this is right on the border of what might make sense to go in topcofea vs topeft. I guess there's nothing actually strictly analysis specific about it (even though right now it's just being used by one analysis) so probably is fine here.

I would pass mass of the Z as an argument though (you could leave 91.2 as the default, as we do in the other pt of the Z mask function), since others might want to use e.g. 91.1876.

**Comment by kmohrman** [comment]
I'm wondering if someone could please comment on the status of this PR? Is it still intended to be merged at some point, or has it been superseded by a different PR in topcoffea and/or topeft?

**Comment by ywan2** [comment]
Hi @kmohrman,

This branch's function is implemented in a new branch. So it's not going to be merged. We can delete it. Thanks for checking!

**Comment by kmohrman** [comment]
Ok, thanks for the update!

Just to confirm, are you ok the PR being closed _and_ its associated branch ([3l-offZ](https://github.com/TopEFT/topcoffea/tree/3l-offZ)) being removed from the topcoffea repository? Or did you still need the branch as a reference?

**Comment by anpicci** [comment]
@kmohrman I would wait for this [PR](https://github.com/TopEFT/topeft/pull/431) to be merged, to avoid that @ywan2 looses any precious information for her work

**Comment by bryates** [comment]
@ywan2 this PR is superseded by #44, right?

**Pull request closed without merging**